zerver/views/message_edit.py

```diff
@@ -113,6 +113,48 @@ def update_message_backend(
     send_notification_to_new_thread: bool = REQ(default=True, validator=check_bool),
     content: Optional[str] = REQ(default=None),
 ) -> HttpResponse:
+    number_changed, message, links_for_embed = check_update_message(
+        user_profile,
+        message_id,
+        stream_id,
+        topic_name,
+        propagate_mode,
+        send_notification_to_old_thread,
+        send_notification_to_new_thread,
+        content,
+    )
+
+    # Include the number of messages changed in the logs
+    request._log_data["extra"] = f"[{number_changed}]"
+    if links_for_embed:
+        event_data = {
+            "message_id": message.id,
+            "message_content": message.content,
+            # The choice of `user_profile.realm_id` rather than
+            # `sender.realm_id` must match the decision made in the
+            # `render_incoming_message` call earlier in this function.
+            "message_realm_id": user_profile.realm_id,
+            "urls": list(links_for_embed),
+        }
+        queue_json_publish("embed_links", event_data)
+    return json_success()
+
+
+def check_update_message(
```

Comment:
I think this should be in `zerver/lib/actions.py`

Reply:
Though I don't see a need for the new function.

Reply:
Because I need to reuse the check and update logic for embedded bots. But yes, I think we can move this to `zerver/lib/actions.py`.